File: route/view_w_raw.py
from .tool.func import *

from .go_api_w_raw import api_w_raw

async def view_w_raw(name = '', rev = '', doc_acl = ''):
    with get_db_connect() as conn:
        rev_str = str(rev)

        sub = '(' + await get_lang('raw') + ')'
        sub += ' (' + rev_str + ')' if rev != '' else ''

        if rev != '':
            menu = [['history_tool/' + rev_str + '/' + url_pas(name), await get_lang('return')]]
        else:
            menu = [['w/' + url_pas(name), await get_lang('return')]]

        data = await api_w_raw(name, rev)
        if data["response"] == "ok":
            data_in = data["data"]
        else:
            data_in = ''

        p_data = ''
        p_data += '''
            <div id="opennamu_preview_area">
                <textarea readonly id="opennamu_edit_textarea" class="opennamu_textarea_500 __ON_TEXTAREA__">''' + html.escape(data_in) + '''</textarea>
            </div>
        '''
        
        if doc_acl == 'on':
            p_data = '' + \
                await get_lang('authority_error') + \
                '<hr class="main_hr">' + \
                p_data
            ''
            
            sub = ' (' + await get_lang('edit') + ')'

        return await render_template(
            name,
            p_data,
            sub,
            menu
        )

File: route/view_w_raw.go
package route

import "opennamu/route/tool"

func View_w_raw(config tool.Config, doc_name string, rev string, do_type string) string {
    db := tool.DB_connect()
    defer tool.DB_close(db)

    sub := "(" + tool.Get_language(db, "raw", true) + ")"
    if rev != "" {
        sub += " (" + rev + ")"
    }

    var menu [][]any

    if rev != "" {
        menu = [][]any{
            { "history_tool/" + tool.Url_parser(rev) + "/" + tool.Url_parser(doc_name), tool.Get_language(db, "return", true) },
        }
    } else {
        menu = [][]any{
            { "w/" + tool.Url_parser(doc_name), tool.Get_language(db, "return", true) },
        }
    }

    data_api_in := ""

    data_api := Api_w_raw(config, doc_name, "", rev)
    if data_api["response"].(string) == "ok" {
        data_api_in = data_api["data"].(string)
    }

    pre_data := `
        <div id="opennamu_preview_area">
            <textarea readonly id="opennamu_edit_textarea" class="opennamu_textarea_500 __ON_TEXTAREA__">` + tool.HTML_escape(data_api_in) + `</textarea>
        </div>
    `

    if do_type == "document_acl" {
        pre_data = `
            ` + tool.Get_language(db, "authority_error", true) + `
            <hr class="main_hr">
        ` + pre_data

        sub = "(" + tool.Get_language(db, "edit", true) + ")"
    }

    out := tool.Get_template(
        db,
        config,
        doc_name,
        pre_data,
        []any{ sub },
        menu,
        map[string]string{},
    )

    return out
}
